app/assets/javascripts/components/assignments/assignment.jsx

```diff
@@ -17,12 +17,10 @@ const Assignment = React.createClass({
   },
   render() {
     const article = this.props.article || CourseUtils.articleFromAssignment(this.props.assignmentGroup[0]);
-
-    article.formatted_title = CourseUtils.formattedArticleTitle(
-        this.props.assignmentGroup[0].language,
-        this.props.assignmentGroup[0].project,
-        article.title,
-        this.props.course.home_wiki);
+    article.formatted_title = CourseUtils.formattedArticleTitle(article);
+    if (!article.formatted_title) {
+      article.formatted_title = CourseUtils.formattedArticleTitle(this.props.assignmentGroup[0]);
```

Comment:
Here you can do `article.formatted_title = CourseUtils.formattedArticleTitle(article, this.course.home_wiki)`

The fewer times we need to call `assignmentGroup[0]`, the more understandable this code will be.